src/Examine/LuceneEngine/Providers/LuceneSearcher.cs

```diff
@@ -291,7 +291,19 @@ public void ScheduleReopen()
                             //if we're on NRT then we should just try to re-open
                             if (_luceneSearcher._nrtWriter != null)
                             {
-                                MaybeReopen();
+                                try
+                                {
+                                    MaybeReopen();
+
+                                }
+                                catch (Exception e)
+                                {
+                                    //It's the initial call to this at the beginning or after successful commit
```

Comment:
This try/catch seems significant, was this caused somewhere during debugging? Should this try/catch be ported individually to another PR?

Reply:
@Shazwazza after refactor it is actually redundant as it was happening only in readonly, but I changed way of handling read only to never occur again, but left that as in time of debugging I hit issue when old reader was already disposed, and new one was not ready yet, and adding that try catch up was fixing about 10-15% of my crashes during debugging :)